lib/matplotlib/tests/test_cbook.py

```diff
@@ -651,3 +651,75 @@ def test_check_shape(target, test_shape):
     with pytest.raises(ValueError,
                        match=error_pattern):
         cbook._check_shape(target, aardvark=data)
+
+
+def test_setattr_cm():
+    class A:
+
+        cls_level = object()
+        override = object()
+        def __init__(self):
+            self.aardvark = 'aardvark'
+            self.override = 'override'
+            self._p = 'p'
+
+        def meth(self):
+            ...
+
+        @property
+        def prop(self):
+            return self._p
+
+        @prop.setter
+        def prop(self, val):
+            self._p = val
+
+    class B(A):
+        ...
+
+    other = A()
+
+    def verify_pre_post_state(obj):
+        # When you access a Python method the function is bound
+        # to the object at access time so you get a new instance
+        # of MethodType every time.
+        #
+        # https://docs.python.org/3/howto/descriptor.html#functions-and-methods
+        assert obj.meth is not obj.meth
+        # normal attribute should give you back the same instance every time
+        assert obj.aardvark is obj.aardvark
+        assert a.aardvark == 'aardvark'
+        # and our property happens to give the same instance every time
+        assert obj.prop is obj.prop
+        assert obj.cls_level is A.cls_level
+        assert obj.override == 'override'
+        assert not hasattr(obj, 'extra')
+        assert obj.prop == 'p'
+        assert obj.monkey == other.meth
+
+    a = B()
+
+    a.monkey = other.meth
+    verify_pre_post_state(a)
+    with cbook._setattr_cm(
+            a, prop='squirrel',
+            aardvark='moose', meth=lambda: None,
+            override='boo', extra='extra',
+            monkey=lambda: None
+
```

Comment:
blank line a bit weird, and I guess you could either add a trailing comma after None or put the closing parentheses on the same line.